code/game/objects/items/weapons/extinguisher.dm

```diff
@@ -148,12 +148,6 @@
 					for(var/atom/atm in get_turf(W))

```

Comment:
there can only be atoms in a turf.
so don't typecast atm in the for header:

```
for(var/a in get_turf(w))
    var/atom/atm = a
```

instead of what's there currently.